nixos/modules/config/nix.nix

```diff
@@ -386,5 +386,11 @@ in
         )
       );
     };
+
+    # Enable git if flake support is enabled
+    programs.git = mkIf (builtins.match ".*flakes.*" != null) {
+      enable = true;
+      package = lib.mkDefault pkgs.gitMinimal;
```

Comment:
Will this change the behavior of manually enabling `programs.git.enable`?

Reply:
If you have flakes enabled, yes.

Reply:
I still don't think git is necessary for flake.

Reply:
It is, files that aren't tracked in git are not part of flakes.

Reply:
The problem is, when people enable flake and enable git, they install `gitMinimal`. If people only enable git, they install `git`:

https://github.com/NixOS/nixpkgs/blob/655a58a72a6601292512670343087c2d75d859c1/nixos/modules/programs/git.nix#L12-L14

Reply:
I guess I could not override the git package.

Reply:
```nix
  mkOptionDefault = mkOverride 1500; # priority of option defaults
  mkDefault = mkOverride 1000; # used in config sections of non-user modules to set a default
  defaultOverridePriority = 100;
  mkImageMediaOverride = mkOverride 60; # image media profiles can be derived by inclusion into host config, hence needing to override host config, but do allow user to mkForce
  mkForce = mkOverride 50;
  mkVMOverride = mkOverride 10; # used by ‘nixos-rebuild build-vm’
```

Reply:
```nix
(builtins.match ".*flakes.*" != null) && !config.programs.git.enable
```

Reply:
Wouldn't that infinitely recurse?

Reply:
You are right, sorry I made a mistake.

Reply:
Mind checking the new version? It should address this concern now.